### PR TITLE
[9.x] Added docs for 'throw' method

### PR DESCRIPTION
**Pull request by ash-jc-allen**
Hi! This PR is only a small one and just adds a small explainer for the `throw` method adding to the `Http` facade today in https://github.com/laravel/framework/pull/41953.

Hopefully, it's all okay and get's the general idea across of how to use the method 😄

### CODE REVIEW

**Comment by taylorotwell** [comment]
Drafting until release day.

File: http-client.md [hunk 1]
```diff
@@ -250,6 +250,17 @@ If you would like to perform some additional logic before the exception is throw
...
     })->json();
 
+Alternatively, if you want to specify before the request is sent that you want to throw an instance of `Illuminate\Http\Client\RequestException` if the response status code indicates a client or server error on the request, you can use the `throw` method on the `PendingRequest` class:
+
+    // Throw an exception if a client or server error occurs...
+    $response = Http::throw()->post(...);
+
+The `throw` method also accepts an optional closure that you can use to perform additional logic if the request encounters a client or server error:
+
+    return Http::throw(function ($response, $e) {
```
Comment (laurentrousseau918) [inline http-client.md:260]: 👍

**Comment by taylorotwell** [comment]
I think I'll just leave this out of the docs for now.

**Comment by ash-jc-allen** [comment]
That's no problem at all :)

**Comment by sebastiaanluca** [comment]
Just my 2 cents, but having this in the documentation would be useful though. I'd rather indicate errors should be thrown before sending a request. Already updated all my code to use the new functionality 👌

**Pull request closed without merging**